sdk/servicebus/Azure.Messaging.ServiceBus/src/Sender/ServiceBusSenderClient.cs

```diff
@@ -0,0 +1,522 @@
+        }
+
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="ServiceBusSenderClient"/> class.
+        /// </summary>
+        ///
+        protected ServiceBusSenderClient()
+        {
+            OwnsConnection = false;
+        }
+
+        /// <summary>
+        ///   Sends a set of events to the associated Service Bus entity using a batched approach.  If the size of events exceed the
+        ///   maximum size of a single batch, an exception will be triggered and the send will fail.
+        /// </summary>
+        ///
+        /// <param name="message">The set of event data to send.</param>
+        /// <param name="cancellationToken">An optional <see cref="CancellationToken"/> instance to signal the request to cancel the operation.</param>
+        ///
+        /// <returns>A task to be resolved on when the operation has completed.</returns>
+        ///
+        /// <seealso cref="SendRangeInternal(IEnumerable{ServiceBusMessage}, CancellationToken)"/>
+        ///
+        public virtual async Task SendAsync(
+            ServiceBusMessage message,
+            CancellationToken cancellationToken = default)
+        {
+            Argument.AssertNotNull(message, nameof(message));
+            await SendRangeAsync(new ServiceBusMessage[] { message }, cancellationToken).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        ///   Sends a set of events to the associated Service Bus entity using a batched approach.  If the size of events exceed the
+        ///   maximum size of a single batch, an exception will be triggered and the send will fail.
+        /// </summary>
+        ///
+        /// <param name="messages">The set of event data to send.</param>
+        /// <param name="cancellationToken">An optional <see cref="CancellationToken"/> instance to signal the request to cancel the operation.</param>
+        ///
+        /// <returns>A task to be resolved on when the operation has completed.</returns>
+        ///
+        /// <seealso cref="SendRangeInternal(IEnumerable{ServiceBusMessage}, CancellationToken)"/>
+        ///
+        public virtual async Task SendRangeAsync(IEnumerable<ServiceBusMessage> messages, CancellationToken cancellationToken = default) =>
+            await SendRangeInternal(messages, cancellationToken).ConfigureAwait(false);
+
+        /// <summary>
+        /// Schedules a message to appear on Service Bus at a later time.
+        /// </summary>
+        /// <param name="message"></param>
+        /// <param name="scheduleEnqueueTimeUtc">The UTC time at which the message should be available for processing</param>
+        /// <param name="cancellationToken"></param>
+        /// <returns>The sequence number of the message that was scheduled.</returns>
+        public virtual async Task<long> ScheduleMessageAsync(ServiceBusMessage message, DateTimeOffset scheduleEnqueueTimeUtc, CancellationToken cancellationToken = default)
+        {
```

Comment:
I think we need to add two more methods one for scheduling array of messages and another one for cancelling those scheduled messages.

1) Schedules given messages to appear on Service Bus Queue/Subscription at a later time
```csharp
public virtual async Task<long[]> ScheduleMessagesAsync(IEnumerable<ServiceBusMessage> messages, DateTimeOffset scheduleEnqueueTimeUtc, CancellationToken cancellationToken = default)
```
2) Cancels multiple messages that were scheduled to appear on a ServiceBus Queue/Subscription.
```csharp
public virtual async Task CancelScheduledMessagesAsync(long[] sequenceNumbers, CancellationToken cancellationToken = default)
```
What do you think?

Reply:
Those seem like sensible methods, but they weren't in Track 1 which makes me think the service doesn't support scheduling in batch. @nemakam

Reply:
Typescript Service Bus SDK supports this feature and was wondering if we could add the same in .Net.

For more context: https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/servicebus/service-bus/src/sender.ts#L174

Reply:
Interesting.. does TS send it as a batch or does it do one message request at a time?
If service supports batching, I'm not sure why we wouldn't want to include this.

Reply:
Looks like TypeScript sends a [single request](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/servicebus/service-bus/src/core/managementClient.ts#L608) for the batch of messages.

Reply:
Nice, in that case we should add it. I will create an issue for this. I'm still curious as to why this isn't in .NET Track 1.

Reply:
Created https://github.com/Azure/azure-sdk-for-net/issues/9874